ssh_key_rotator/publisher/key_publisher.py

```diff
@@ -0,0 +1,85 @@
+from abc import ABC, abstractmethod
+import os
+import shutil
+import sys
+import boto3
+from ssh_key_rotator.custom_keygen import (
+    generate_private_public_key_in_file,
+    generate_private_public_key,
+)
+from ssh_key_rotator.util import get_user_path, get_username
+
+
+class Publisher(ABC):
+    """Abstract key publisher base class"""
+
+    @abstractmethod
+    def publish_new_key(self) -> str:
+        """Abstract method for publishing a new public key"""
+
+
+class S3Publisher(Publisher):
+    """S3 Publisher class"""
+
+    def __init__(self, bucket_name: str, host: str, user_id: str):
+        self.bucket_name = bucket_name
+        self.host = host
+        self.user_id = user_id
+
+    def publish_new_key(self) -> str:
+        # Generate new public/private key pair
+        private_key, public_key = generate_private_public_key()
+
+        # Store the new public key in S3 bucket
+        s3_client = boto3.client("s3")
+        s3_client.put_object(
+            Body=public_key,
+            Bucket=self.bucket_name,
+            Key=f"{self.host}/{self.user_id}-cert.pub",
+        )
+
+        # Store the private key on the local machine
+        private_key_path = f"{get_user_path()}/.ssh/{self.host}/{self.user_id}"
```

Comment:
we should be using a temp location

Reply:
We will include that functionality and allow for a different directory to be passed in. The DataStore already takes care of the temporary bit